docs/guide/custom-ui.md

```diff
@@ -1,39 +1,292 @@
+    add: {
+      rating: {
+        type: 'starRating',
+        label: 'Star Rating',
+        required: true
+      }
+    }
+  },
+  columns: {
+    add: {
+      rating: {
+        label: 'Rating',
+        component: 'ColumnStarRating'
+      }
+    }
+  }
+};
+```
+  <template v-slot:caption>
+    modules/review/index.js
+  </template>
+</AposCodeBlock>
+
+### Implementing the browser-side part
+
+This code makes reference to a Vue component, `ColumnStarRating`, that doesn't exist yet. Next we'll introduce that component:
+
+<AposCodeBlock>
+```js
+<template>
+  <p
+    class="apos-table__cell-field"
+    :class="`apos-table__cell-field--${header.name}`"
+  >
+    <span v-if="hasValue">
+      <span v-for="index in 5" :key="index" class="rating">{{ isActive(index) ? '☆' : '★' }}</span>
+    </span>
+    <span v-else>
+      None
+    </span>
+  </p>
+</template>
+
+<script>
+import AposCellMixin from 'Modules/@apostrophecms/ui/mixins/AposCellMixin';
+
+export default {
+  name: 'AposCellBasic',
+  mixins: [ AposCellMixin ],
+  computed: {
+    hasValue() {
+      return this.get(this.header.name) != null;
+    }
+  },
+  methods: {
+    isActive(index) {
+      return index <= (this.get(this.header.name) || 0);
+    }
+  }
+};
+</script>
+
+```
+  <template v-slot:caption>
+    modules/star-range-field/ui/apos/components/ColumnStarRating.vue
+  </template>
+</AposCodeBlock>
+
+While the Vue component could be placed in any module, it makes the most sense to group it in the `star-range-field` module because it might be used as a column in any piece type, not just a `review`.
+
+This component uses `AposCellMixin` to do two important things:
+
+* The component gains access to the configuration of the column by accessing the `header` prop, which is automatically declared by the mixin.
+* The component can fetch properties from the piece by invoking `this.get` with any property name. Following Apostrophe's convention this method automatically fetches from the published version of the piece if it exists, otherwise from the draft.
+
+## Overriding standard Vue.js components by name
+
+Most of the time we don't need to override admin UI components that ship with Apostrophe. But if we have a need, we can do so by **placing a file with the same name as a standard component in the `ui/apos/components` subdirectory of a project-level module.** You can also do this in a custom npm module to achieve reuse across projects.
+
+Apostrophe will use only the last version of a component that it finds during startup. The general startup order is:
+
+1. Core Apostrophe modules
+2. Installed and project-level modules, in the order they are configured in `app.js`
+that is configured. For instance, if the last module in our project's `app.js` modules list contains a `ui/apos/components/AposLogPadless.vue` file, that logo will be used in the admin bar, in place of the version that is normally loaded from Apostrophe core or in any module configured earlier.
```

Comment:
There's odd line breaking here. The list is useful for quick scanning, so the "For instance" should be the start of a new paragraph. The "that is configured" can be back with the previous line.

For doc review and diffing it's helpful to avoid extra line breaks.

Reply:
Actually "that is configured." seems like it should be deleted.